api/utils/error_parsing.ts

```diff
@@ -0,0 +1,100 @@
+interface IErrorLocation {
```

Comment:
for all of these parsers: should their bodies be wrapped in try/catch clauses just in case some string splitting/matching/etc breaks for some reason? (so that we always return the original error)

Reply:
sure